api/v1alpha1/apidefinition_webhook_test.go

```diff
@@ -80,3 +81,125 @@ func TestApiDefinition_Default_DoNotTrack(t *testing.T) {
 		t.Fatalf("expected DoNotTrack to be false as explicitly set, got %v", *in.Spec.DoNotTrack)
 	}
 }
+
+func TestApiDefinition_validateTarget(t *testing.T) {
+	invalidRewriteApiDef := ApiDefinition{
+		Spec: APIDefinitionSpec{
+			APIDefinitionSpec: model.APIDefinitionSpec{
+				Proxy: model.Proxy{TargetURL: "/test"},
+				VersionData: model.VersionData{
+					Versions: map[string]model.VersionInfo{
+						"Default": {
+							ExtendedPaths: &model.ExtendedPathsSet{
+								URLRewrite: []model.URLRewriteMeta{{RewriteTo: ""}},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	invalidTriggerApiDef := ApiDefinition{
+		Spec: APIDefinitionSpec{
+			APIDefinitionSpec: model.APIDefinitionSpec{
+				Proxy: model.Proxy{TargetURL: "/test"},
+				VersionData: model.VersionData{
+					Versions: map[string]model.VersionInfo{
+						"Default": {
+							ExtendedPaths: &model.ExtendedPathsSet{
+								URLRewrite: []model.URLRewriteMeta{
+									{RewriteTo: "", Triggers: []model.RoutingTrigger{{}}},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	errorDetail := "can't be empty"
+	missingTargetURLErr := field.Error{
+		Type:   field.ErrorTypeRequired,
+		Field:  path("proxy", "target_url").String(),
+		Detail: errorDetail,
+	}
+
+	cases := []struct {
+		name        string
+		apiDef      ApiDefinition
+		expectedErr *field.Error
+	}{
+		{
+			name: "valid ApiDefinition",
+			apiDef: ApiDefinition{
+				Spec: APIDefinitionSpec{
+					APIDefinitionSpec: model.APIDefinitionSpec{
+						Proxy: model.Proxy{TargetURL: "/test"},
+					},
+				},
+			},
+			expectedErr: nil,
+		},
+		{
+			name:        "invalid ApiDefinition, missing proxy.target_url",
```

Comment:
@buraksekili Can you add a test where `TargetURL` is set to "" ?
Also when `TargetInternal` is set and `TargetURL` is not.

Reply:
Thanks @komalsukhani. I added new test cases to cover the scenarios that you mentioned.